build-in-container/Dockerfile

```diff
@@ -42,5 +42,3 @@ RUN tdnf update -qy tdnf && \
                 rsync \
                 tar \
                 wget
-
-RUN echo "source /mariner/scripts/setup.sh" >> /root/.bashrc
```

Comment:
I refactored the container creation part because now we have multiple optional arguments, which need to be passed to a starter script inside the container. It's much cleaner now, and would be easier to work with as we do more development. Earlier I was using different entrypoints for different types of container (build v/s interactive). Now, it's the same entrypoint, with conditional elements based on container_type

Reply:
This refactor should be its own PR, separate from the individual features added. Hard to make assessments on if the refactor yields a cleaner result when the refactor is tied in with multiple features.

Reply:
Sure, created a new PR for the refactor changes: https://github.com/microsoft/CBL-MarinerTutorials/pull/45/files